src/MongoDB.Bson/ObjectModel/ObjectId.cs

```diff
@@ -88,7 +90,9 @@ public ObjectId(DateTime timestamp, int machine, short pid, int increment)
         /// <param name="machine">The machine hash.</param>
         /// <param name="pid">The PID.</param>
         /// <param name="increment">The increment.</param>
-        public ObjectId(int timestamp, int machine, short pid, int increment)
+#pragma warning disable CS3001
+        public ObjectId(uint timestamp, int machine, short pid, int increment)
```

Comment:
I noticed that spec requires that `_b`(Random) value should be calculated based on:

    combining time, process ID, and hostname.

currently, we don't include `time` in the calculation logic

Reply:
Then, I'm not sure about this test:
    
    Ensure that after a new process is created through a fork() or similar process creation operation, the "random number unique to a machine and process" is no longer the same as the parent process that created the new process

I expected that in case if we don't make proposed `fork` logic, a random value should be the same for two consecutive calls of `ObjectId.GenerateNewId()`. In case if we add a time, I guess the values will be different
cc: @jyemin , @rstam

Reply:
machine and pid values are calculated here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/85/files#diff-312b5ea5d26aab247a43a999283d4b53R34-R35